### ISSUE TITLE
Consuming version of API

### ISSUE THREAD

**Issue by vorner** [alexcrichton/tokio-process#15]
The current API, like `.spawn_async` borrow the command. This is nice if I want to start multiple similar commands from the same builder.

However, in case I want to start bunch of processes, a consuming API would work better. I mean something like this (it won't probably compile for more reasons than just the borrow checker, but it should illustrate the intention):

```rust
let stuff: Vec<Something> = prepare_data();
let commands = stream::iter(stuff.into_iter().map(Ok::<_, io::Error>))
    .map(|something| {
        Command::new("process").arg(something).spawn_async(&handle)
    })
    .buffer_unordered(10) // Run up to 10 commands in parallel
    .for_each(|_| Ok(());
core.wait(commands).unwrap();
```

This doesn't work because the returned future borrows the command and it is dropped at the end of the `map`.

I believe this might be quite reasonable use, but there's no obvious (to me) way to do it now.

**Comment by alexcrichton** [comment]
Hm I'm not sure I quite follow, the `Child` struct return shouldn't borrow the `Command`? What's the error you're running into?

**Comment by vorner** [comment]
I postponed the creation of the issue and I no longer have the code that triggered it. However, now I can't reproduce it and I just removed the workaround from the code and it compiles. I must have mis-analyzed some other unrelated borrow checker error and blamed it on the wrong culprit.

My apologies,  I'll save the exact code and error next time I think something doesn't work O:-).

**Comment by alexcrichton** [comment]
No worries!
